src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -1587,13 +1629,49 @@ codet java_bytecode_convert_methodt::convert_instructions(
     }
     else if(statement=="iinc")
     {
+      code_blockt block;
+      block.add_source_location()=i_it->source_location;
+      // search variable on stack
+      const exprt &locvar=variable(arg0, 'i', i_it->address, CAST_AS_NEEDED);
+      for(size_t i=0; i<stack.size(); i++)
+      {
+        exprt &stack_entry=stack[i];
+        if(stack_entry.id()==ID_typecast)
+        {
+          const typecast_exprt &cast=to_typecast_expr(stack_entry);
+          stack_entry=cast.op();
+        }
+        // is the stack entry a local variable?
+        if(stack_entry.id()!=ID_symbol)
+          continue;
+        const symbol_exprt &entry=to_symbol_expr(stack_entry);
+
+        irep_idt loc_var_id;
+        if(locvar.id()==ID_typecast)
+          loc_var_id=to_symbol_expr(to_typecast_expr(locvar)).get_identifier();
+        else
+          loc_var_id=to_symbol_expr(locvar).get_identifier();
+        // if the local variable for iinc is on the stack
+        // save value in temp var to preserve it and
+        // change stack entry to point to temp var
+        if(entry.get_identifier()==loc_var_id)
+        {
+          const exprt tmp_var=
+            tmp_variable("stack_iinc_lvar", java_int_type());
+          block.copy_to_operands(
+            code_assignt(tmp_var, locvar));
```

Comment:
Make that code more similar to the above by introducing a temporary variable, which will then also have the source location assigned.